GithubProfileWiki/Screens/FollowersList/FollowersListViewController.swift

```diff
@@ -9,12 +9,38 @@ import UIKit
 
 class FollowersListViewController: UIViewController {
     var username: String?
+    var followerListViewModel = FollowersListViewModel()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
         view.backgroundColor = .systemBackground
-        navigationController?.isNavigationBarHidden = false
         navigationController?.navigationBar.prefersLargeTitles = true
         title = username
     }
+    
+    init(userName: String) {
+        super.init(nibName: nil, bundle: nil)
+        self.username = userName
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: true)
+        
+        followerListViewModel.fetchFollowers(userName: username ?? "", param: [:]) { (model, error) in
+            if let _ = error {
+                self.presentAlertPopupOnMainThread(title: "Error", message: error?.localizedDescription ?? "Hata", buttonTitle: "Close")
```

Comment:
you can call completion handler in main thread in network layer

Reply:
The reason behind this; 

okay, I made the network calls on the background thread however after it is completed I have to update the UI with the data. So all the UI operations should be on the main thread. Since I was in the background thread then I have to use DispatchQueue.main.async {} for completion part to make changes on UI right?

I am asking it because I know completionHandlers are async so why do I have to move main thread and declare them async again?